File: project_euler/Problem045/Problem045.java
package project_euler;

public class Problem045 {
    public void solution() {
        long index = 144L;
        while (true) {
            long hexagon = index * ((index << 1) - 1);
            if (isPentagon(hexagon)) {
                System.out.println(hexagon);
                return;
            }
            ++index;
        }
    }

    private boolean isPentagon(long value) {
        double auxiliary = Math.sqrt(value * 24 + 1);
        return Math.abs(Math.floor(auxiliary + 0.5) - auxiliary) < 0.00000001 && 0 == ((long) auxiliary + 1) % 6;
    }
}

File: project_euler/Problem045/Problem045.py
import math

class Problem045:
    def solution(self) -> None:
        index = 144
        while True:
            hexagon = index * ((index << 1) - 1)
            if self.__is_pentagon(hexagon):
                print(hexagon)
                return
            index += 1

    def __is_pentagon(self, value: int) -> bool:
        auxiliary = math.sqrt(value * 24 + 1)
        return abs(math.floor(auxiliary + 0.5) - auxiliary) < 0.00000001 and not (int(auxiliary) + 1) % 6
